File: app/src/main/java/mono/hg/models/WebSearchProvider.java
package mono.hg.models;

import androidx.annotation.Nullable;

public class WebSearchProvider {
    private String url;
    private String name;
    private String id;

    public WebSearchProvider(String name, String url) {
        this.url = url;
        this.name = name;
        this.id = name;
    }

    public WebSearchProvider(String name, String url, String id) {
        this.url = url;
        this.name = name;
        this.id = id;
    }

    public String getUrl() {
        return url;
    }

    public String getName() {
        return name;
    }

    public String getId() {
        return id;
    }

    @Override public boolean equals(@Nullable Object obj) {
        WebSearchProvider object = (WebSearchProvider) obj;

        // URL can be shared, but names should stay unique.
        if (object != null) {
            return this.getName().equals(object.getName()) || this.getClass() == obj.getClass();
        } else {
            return false;
        }
    }
}

File: app/src/main/java/mono/hg/models/WebSearchProvider.kt
package mono.hg.models

class WebSearchProvider {
    var url: String?
        private set
    var name: String?
        private set
    var id: String?
        private set

    constructor(name: String?, url: String?) {
        this.url = url
        this.name = name
        id = name
    }

    constructor(name: String?, url: String?, id: String?) {
        this.url = url
        this.name = name
        this.id = id
    }

    override fun equals(other: Any?): Boolean {
        val `object` = other as WebSearchProvider?

        // URL can be shared, but names should stay unique.
        return if (`object` != null) {
            name == `object`.name || this.javaClass == other!!.javaClass
        } else {
            false
        }
    }
    override fun hashCode(): Int {
        var result = url?.hashCode() ?: 0
        result = 31 * result + (id?.hashCode() ?: 0)
        return result
    }
}